src/core/Utils.re

```diff
@@ -1,28 +1,2 @@
 let foi = float_of_int;
-let iof = int_of_float;
-
-type scanfState = {
```

Comment:
I typically don't like to delete code until later, but I actually think this version of `scan` / `reduce` is pretty meh. It's also not currently in use, so I decided to scrap it for now with the idea that we may be able to improve it later if we need it.